wallet/src/main/scala/org/bitcoins/wallet/Wallet.scala

```diff
@@ -98,6 +98,45 @@ sealed abstract class Wallet
     }
   }
 
+  override def createNewAccount(): Future[WalletApi] =
+    createNewAccount(DEFAULT_HD_COIN.purpose)
+
+  // todo: check if there's addresses in the most recent
+  // account before creating new
+  override def createNewAccount(purpose: HDPurpose): Future[Wallet] = {
+
+    accountDAO
+      .findAll()
+      .map(_.filter(_.hdAccount.purpose == purpose))
+      .map(_.sortBy(_.hdAccount.index))
+      .map(_.lastOption)
+      .flatMap { mostRecentOpt =>
+        val accountIndex = mostRecentOpt match {
+          case None          => 0 // no accounts present in wallet
+          case Some(account) => account.hdAccount.index + 1
+        }
+        logger.debug(
```

Comment:
This is probably worth a info log no?

Reply:
It gets pretty noisy in tests, but I guess the better solution there is to turn log levels down in tests